libraries/lib-track/Track.h

```diff
@@ -519,13 +522,19 @@ class TRACK_API Track /* not final */
 
    // Return true iff the attribute is recognized.
    bool HandleCommonXMLAttribute(const std::string_view& attr, const XMLAttributeValueView& valueView);
+
+private:
+   virtual void DoOnProjectTempoChange(
+      const std::optional<double>& oldTempo, double newTempo) = 0;
+
+   std::optional<double> mProjectTempo;
```

Comment:
The previous solution remembered old project tempo in ProjectTimeSignature.  Now you store last known tempo in each track.

I think I understand what justifies this.  It's not always duplicated everywhere.  A track could be duplicated into an undo state or in the clipboard.  It remembers what project tempo was at the time the duplicate track was created.  We don't want to normalize it to some tempo value, because we want to delay the time when we must compute the time stretch, which is not perfectly invertible, and so, lossy.

Reply:
This, and also to reduce cognitive load when trying to figure out what happens when tracks in the undo stack miss out on project tempo updates and later get inserted into the list. Now it's easier.